src/SocketEvents.ts

```diff
@@ -0,0 +1,352 @@
+		objPart: ioBroker.PartialObject,
+		callback?: ErrorAsString<ioBroker.SetObjectCallback>,
+	): void;
+	getObject(
+		id: string,
+		callback?: ErrorAsString<ioBroker.GetObjectCallback>,
+	): void;
+	getForeignObjects:
+		| ((
+				pattern: string,
+				type: ioBroker.ObjectType,
+				callback: ErrorAsString<ioBroker.GetObjectsCallback>,
+		  ) => void)
+		| ((
+				pattern: string,
+				callback: ErrorAsString<ioBroker.GetObjectsCallback>,
+		  ) => void);
+
+	getStates(callback?: ErrorAsString<ioBroker.GetStatesCallback>): void;
+	getForeignStates(
+		pattern: string,
+		callback?: ErrorAsString<ioBroker.GetStatesCallback>,
+	): void;
+
+	getState(
+		id: string,
+		callback?: ErrorAsString<ioBroker.GetStateCallback>,
+	): void;
+	getBinaryState(id: string, callback?: GetBinaryStateCallback): void;
+
+	setState(
+		id: string,
+		val: ioBroker.State | ioBroker.StateValue | ioBroker.SettableState,
+		callback?: ErrorCallback,
+	): void;
+	setBinaryState(id: string, base64: string, callback?: ErrorCallback): void;
+
+	readDir(
+		adapterName: string | null,
+		path: string,
+		callback: ErrorAsString<ioBroker.ReadDirCallback>,
+	): void;
+	readFile(
+		adapterName: string | null,
+		path: string,
+		callback: ErrorAsString<ioBroker.ReadFileCallback>,
+	): void;
+	readFile64(
+		adapterName: string | null,
+		path: string,
+		callback: ReadFile64Callback,
+	): void;
+	writeFile(
+		adapterName: string | null,
+		path: string,
+		data: string,
+		callback: ErrorCallback,
+	): void;
+	writeFile64(
+		adapterName: string | null,
+		path: string,
+		dataBase64: string,
+		callback: ErrorCallback,
+	): void;
+	deleteFile(
+		adapterName: string | null,
+		path: string,
+		callback: ErrorCallback,
+	): void;
+	deleteFolder(
+		adapterName: string | null,
+		path: string,
+		callback: ErrorCallback,
+	): void;
+	fileExists(
+		adapterName: string | null,
+		path: string,
+		callback: GenericCallback<boolean>,
+	): void;
+
+	getHistory(
+		id: string,
+		options: ioBroker.GetHistoryOptions,
+		callback: ErrorAsString<ioBroker.GetHistoryCallback>,
+	): void;
+
+	getVersion(callback: GetVersionCallback): void;
+	getAdapterName(callback: GetAdapterNameCallback): void;
+
+	getCompactSystemConfig(
+		callback: ErrorAsString<ioBroker.GetObjectCallback<"system.config">>,
+	): void;
+	checkFeatureSupported(
+		featureName: string,
+		callback: GenericCallback<boolean>,
+	): void;
+
+	sendTo(
+		instance: string,
+		command: string,
+		data: ioBroker.MessagePayload,
+		// TODO: What is the actual type of the returned value here?
+		callback?: (result: any) => void,
+	): void;
+	cmdExec(
+		hostName: string,
+		commandId: string,
+		command: string,
+		callback?: ErrorCallback,
+	): void;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type AdminListenEvents = {};
+
+export interface AdminEmitEvents {
+	sendToHost(
+		hostname: string,
+		command: string,
+		message: any,
+		// TODO: What is the actual type of the returned value here?
```

Comment:
sendToHost should be generic in my opinion (with a default type of "any" for convenience), because you know what your will get back from the adapter when calling it.
```js
sendToHost<T = any>(
		hostname: string,
		command: string,
		message: any,
                callback?: (result: T) => void,
	): void;
```

Reply:
Agreed!